### PR TITLE
Adding the mobile card styles example and documentation.

### LINKED ISSUE #1235: Combine dashboard theme table mobile card pattern

### PR DESCRIPTION
**Pull request by richmccartney**
## Done

- Port over the new table mobile card pattern
- Create documentation for the pattern and an example

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/
- View the pattern at http://0.0.0.0:8101/vanilla-framework/examples/patterns/table-mobile-card/table-mobile-card/

## Details

Fixes #1235

### CODE REVIEW

File: docs/en/patterns/table-mobile-card.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+---
+
+# Table mobile card
+
+Applying the class `.p-table--mobile-card` will give any table a new responsive card view when on smaller screens. Each cell will require
+an `[aria-label]` to describe the cell on a mobile screen. We use the content to create a pseudo element and keep it line with the content.
+
+The `<thead>` element is completely hidden from  view on a smaller screen and if the table holds a `.p-contextual-menu` pattern all the children elements will be visible and be interactive.
+
+<a href="https://vanilla-framework.github.io/vanilla-framework/examples/patterns/table-mobile-card/"
```
Comment (anthonydillon) [inline docs/en/patterns/table-mobile-card.md]: The example URL is: `examples/patterns/table-mobile-card/table-mobile-card`. You should only create a new directory if you have more then one example.

File: examples/patterns/table-mobile-card/table-mobile-card.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      <th scope="col" role="columnheader" id="t-name" aria-sort="none">Name</th>
+      <th scope="col" role="columnheader" id="t-users" aria-sort="none" class="u-align--right">Users</th>
+      <th scope="col" role="columnheader" id="t-units" aria-sort="none" class="u-align--right">Units</th>
+      <th scope="col" role="columnheader" id="t-revenue" aria-sort="none" class="u-align--right">Actions</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr role="row">
+      <td role="rowheader" aria-label="Name">Grape</td>
+      <td role="gridcell" aria-label="Users"class="u-align--right">8</td>
```
Comment (anthonydillon) [inline examples/patterns/table-mobile-card/table-mobile-card.html]: Space missing before class.

File: examples/patterns/table-mobile-card/table-mobile-card.html [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      <td role="gridcell" aria-label="Units" class="u-align--right">19</td>
+      <td role="gridcell" class="u-align--right">
+        <span class="p-contextual-menu p-contextual-menu--left" role="menubar">
```
Comment (anthonydillon) [inline examples/patterns/table-mobile-card/table-mobile-card.html]: It not a good idea to nest patterns if you can help it. If the pattern changes to create multiple places to update the example. Feel free to document the usage if you feel it's useful.

**Comment by anthonydillon** [comment]
## QA issues:
The menu is not collapsed on large screen
table mobile card - vanilla framework - examples 1

The delete button has no top margin on small screens:
table mobile card - vanilla framework - examples

File: docs/en/patterns/table-mobile-card.md (hunk 1 above)
Comment (richmccartney) [inline docs/en/patterns/table-mobile-card.md]: I've renamed this as it should be under a tables folder, there are two more table patterns coming.

**Comment by richmccartney** [comment]
@anthonydillon I've removed the context menu now and fixed any other issues.

**Comment by richmccartney** [comment]
@anthonydillon done.

**Comment by kwm14** [comment]
Looking good 👍

**Pull request merged**